jerry-core/parser/regexp/re-parser.cpp

```diff
@@ -619,19 +623,26 @@ re_parse_next_token (re_parser_ctx_t *parser_ctx_p, /**< RegExp parser context *
       }
       else if (ch == LIT_CHAR_LOWERCASE_C)
       {
-        if (lit_utf8_iterator_is_eos (iter_p))
+        if (!lit_utf8_iterator_is_eos (iter_p))
         {
-          out_token_p->value = ch;
-          break;
-        }
-
-        ch = lit_utf8_iterator_peek_next (iter_p);
+          ch = lit_utf8_iterator_peek_next (iter_p);
 
-        if ((ch >= LIT_CHAR_ASCII_UPPERCASE_LETTERS_BEGIN && ch <= LIT_CHAR_ASCII_UPPERCASE_LETTERS_END)
-            || (ch >= LIT_CHAR_ASCII_LOWERCASE_LETTERS_BEGIN && ch <= LIT_CHAR_ASCII_LOWERCASE_LETTERS_END))
+          if ((ch >= LIT_CHAR_ASCII_UPPERCASE_LETTERS_BEGIN && ch <= LIT_CHAR_ASCII_UPPERCASE_LETTERS_END)
+              || (ch >= LIT_CHAR_ASCII_LOWERCASE_LETTERS_BEGIN && ch <= LIT_CHAR_ASCII_LOWERCASE_LETTERS_END))
+          {
+            out_token_p->value = (ch % 32);
+            lit_utf8_iterator_advance (iter_p, 1);
```

Comment:
Please, change this call to `lit_utf8_iterator_incr (iter_p)`, the same effect, but simpler.